src/pheval/utils/semsim_utils.py

```diff
@@ -1,66 +1,129 @@
 """
 Contains all pheval utility methods
 """
+import logging
+from itertools import combinations
 from pathlib import Path
+from typing import List
 
-import numpy
 import pandas as pd
 import plotly.express as px
+import polars as pl
+import seaborn as sns
+from matplotlib import pyplot as plt
 
 import pheval.utils.file_utils as file_utils
 
+info_log = logging.getLogger("info")
 
-def filter_non_0_score(data: pd.DataFrame, col: str) -> pd.DataFrame:
+
+def semsim_comparison(input: List[Path], score_column: str, analysis: str, output: Path):
+    for s in set(combinations(input, 2)):
+        semsim_left = s[0]
+        semsim_right = s[1]
+        if analysis == "heatmap":
+            semsim_heatmap_plot(semsim_left, semsim_right, score_column)
+        if analysis == "percentage_diff":
+            percentage_diff(semsim_left, semsim_right, score_column, output)
+    if analysis == "distribution":
+        semsim_score_distribution_plot(input, score_column, output)
+
+
+def filter_non_0_score(data: pl.DataFrame, col: str) -> pd.DataFrame:
```

Comment:
Have you got tests for all these other semsim util functions?

Reply:
I'll send in the next commit

Reply:
I can see that there is a test for the CLI command in `test_cli.py`, what I meant is do you have tests for all the smaller methods that you are calling? Like the `diff_semsim` method in `semsim_utils.py`